packages/block-editor/src/components/inserter/style.scss

```diff
@@ -14,43 +15,107 @@ $block-inserter-search-height: 38px;
 	@include break-medium {
 		position: relative;
 	}
-}
 
-.block-editor-inserter__popover > .components-popover__content {
-	@include break-medium {
-		overflow-y: visible;
-		height: $block-inserter-content-height + $block-inserter-tabs-height + $block-inserter-search-height;
+	.block-editor-inserter__toggle {
+		display: inline-flex;
+		align-items: center;
+		color: $dark-gray-500;
+		background: none;
+		cursor: pointer;
+		border: none;
+		outline: none;
+		transition: color 0.2s ease;
+		@include reduce-motion("transition");
 	}
 }
 
-.block-editor-inserter__toggle {
-	transition: color 0.2s ease;
-	@include reduce-motion("transition");
+// Popover container.
+.block-editor-inserter__popover {
+
+	// The in-canvas appender library.
+	> .components-popover__content {
+		height: 440px;
+	}
+
+	// The toolbar library.
+	&.is-top-toolbar-inserter > .components-popover__content {
+		@include break-medium {
+			overflow-y: visible;
+			height: 92vh; // This is limited by a max-height from the popover component itself.
+			width: 82vw; // This is limited by a max-width.
+
+			.is-fullscreen-mode & {
+				width: 86vw;
+			}
+
+			// We take up the whole space horizontally, so as to afford a preview panel appearing.
+			// The layer is only for layout, and should not block clicks through it.
+			pointer-events: none;
+		}
+
+		// Remove the popover styling and apply it instead to each sub panel.
```

Comment:
Another option is to consider the preview as an absolute positioned div outside of the flow of the popover. It might be simpler.